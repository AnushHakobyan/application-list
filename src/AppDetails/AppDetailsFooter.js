import React from 'react';
import PropTypes from 'prop-types';
import { subscriptionShape } from '../utils/typeShapes';

const AppDetailsFooter = ({ subscriptions }) => (
  <div className="box-info--footer">
    <ul>
      {subscriptions.map(({ name, price }) => (
        <li key={name} data-testid={name}>
          <span>{name}</span>
          {' '}
          <h3>{price  || 'free'}<sup>{price ? 'â‚¬' : ''}</sup></h3>
        </li>
      ))}
    </ul>
  </div>
);

AppDetailsFooter.propTypes = {
  subscriptions: PropTypes.arrayOf(subscriptionShape),
};

export default AppDetailsFooter;