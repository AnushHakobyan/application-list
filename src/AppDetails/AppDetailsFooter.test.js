import AppDetailsFooter from './AppDetailsFooter';

const subscriptions = [
  { name: 'trial', price: 0 },
  { name: 'starter', price: 100 },
  { name: 'advanced', price: 200 },
]

describe ('AppDetailsFooter', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <AppDetailsFooter subscriptions={subscriptions} />
    );
  }),
  it ('should render subscriptions correctly', () => {
    expect(wrapper).toMatchSnapshot();
  }),
  it ('should render \'free\' if price is 0 ', () => {
    expect(wrapper.find('[data-testid="trial"] h3').text()).toEqual('free');
  })
})