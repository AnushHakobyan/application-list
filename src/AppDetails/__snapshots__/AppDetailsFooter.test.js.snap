// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDetailsFooter should render subscriptions correctly 1`] = `
<div
  className="box-info--footer"
>
  <ul>
    <li
      data-testid="trial"
      key="trial"
    >
      <span>
        trial
      </span>
       
      <h3>
        free
        <sup />
      </h3>
    </li>
    <li
      data-testid="starter"
      key="starter"
    >
      <span>
        starter
      </span>
       
      <h3>
        100
        <sup>
          €
        </sup>
      </h3>
    </li>
    <li
      data-testid="advanced"
      key="advanced"
    >
      <span>
        advanced
      </span>
       
      <h3>
        200
        <sup>
          €
        </sup>
      </h3>
    </li>
  </ul>
</div>
`;
